sgt-back-end/index.js

```diff
@@ -0,0 +1,155 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+app.use(express.json());
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({ error: 'gradeId must be a positive integer' });
+  }
+  const sql = `
+    select "gradeId",
+            "name",
+            "course",
+            "score",
+            "createdAt"
+      from "grades"
+      where "gradeId" = $1
+      `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({ error: `Cannot find grade with gradeId ${gradeId}` });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select "gradeId",
+            "name",
+            "course",
+            "score",
+            "createdAt"
+      from "grades"
+      `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const newEntry = req.body;
+  for (const property in newEntry) {
+    if (newEntry[property] === undefined || newEntry[property].length === 0) {
+      return res.status(400).json({ error: `${property} is a required field.` });
+    }
+  }
+  newEntry.score = Number(newEntry.score);
+  if (!Number.isInteger(newEntry.score) || newEntry.score > 100 || newEntry.score < 0) {
+    return res.status(400).json({ error: 'score must be an integer between 0 - 100' });
+  }
```

Comment:
What happens if the request body only has a score and nothing else?

Reply:
Adding screenshot showing attempt to post with score only, the first if-statement checking for required fields still returns 400 error.
![sgt-back-end-score-only-post](https://user-images.githubusercontent.com/64382900/145520981-53c8ede6-1901-423f-88f6-35193697fbe6.JPG)